rsconnect/actions_content.py

```diff
@@ -436,7 +436,7 @@ def apply_content_type_filter(item: ContentItemV1):
     def apply_version_filter(items: Iterator[ContentItemV1], version_filter: VersionSearchFilter):
         def do_filter(item: ContentItemV1) -> bool:
             vers = None
-            if version_filter.name in item:
+            if version_filter.name not in item:
                 return False
```

Comment:
For comparison, see the code from the 1.23.0 release:

https://github.com/rstudio/rsconnect-python/blob/1.23.0/rsconnect/actions_content.py#L393